src/hooks/useDatePicker.spec.tsx

```diff
@@ -32,7 +40,7 @@ describe('useDatePicker hook', () => {
             const options = {
                 locale: 'en-GB',
                 timeZone: 'Africa/Khartoum',
-                // no calendar means it should default to gregory
+                calendar: 'gregory' as const,
```

Comment:
This additional option makes me think that it is now required to explicitly pass a `calendar` string. Which I do not think we would want? The `useResolvedLocaleOptions` should ensure some default value is returned for all relevant properties.

Reply:
OK, I feel there are separate unrelated valid points which require further discussion:
1. The philosophical point: I don't think that the fact that this is a component which specifically offers multi-calendar support is a reason to not provide a default calendar system. Rather than being philosophical I'd like to propose we act pragmatic. DHIS2 is used in dozens of countries and the vast majority is all on the same calendar system, so let's default to that.
1. Using a fallback value from resolvedOptions VS a hardcoded string: I think you make a good point there. The only useful values here are the instance's calendar system and some sane default. The user's calendar system is not that relevant.
1. Defaulting to `iso8601` or `gregory`: I don't think that for the date picker it matters, but for the period generation it does make a difference in terms of when things start. Our week number logic is rooted in `iso8601`, so let's default to that.
1. Validating the provided calendar system: I think it's fine if this blows up when a invalid calendar is being passed to the hook/component. Not sure if the resulting error message is clear enough as it is, or if we should make an effort to catch it and throw a clearer error. I'd say this can be left out of scope for now.